src/Diagram/FlowChart.hs

```diff
@@ -239,14 +228,14 @@ pMultiNode nodes = do
       node <- pNode
       pMultiNode $ (++) nodes [node]
 
-pFlowChartGraphInit :: Parser [FlowChartGraph]
-pFlowChartGraphInit = do
+pSubFlowChartGraphInit :: Parser [SubFlowChartGraph]
+pSubFlowChartGraphInit = do
   nodes <- pMultiNodeInit
   let isolatedNodes = vertices nodes
-  pFlowChartGraph [isolatedNodes]
+  pSubFlowChartGraph [isolatedNodes]
 
-pFlowChartGraph :: [FlowChartGraph] -> Parser [FlowChartGraph]
-pFlowChartGraph graphs = do
+pSubFlowChartGraph :: [SubFlowChartGraph] -> Parser [SubFlowChartGraph]
```

Comment:
Hmm, not sure what's going on here. A parser that takes the very thing it parses as its input? Can we refactor this to not do that?

Reply:
Sub flow chart structure should probably be a type of its own based on `Data.Tree`? So that we can leave the graph type as it is simple without encoding sub-flow-chart structure into it. 

(Haven't looked at this in detail).